ecmd-core/cmd/ecmdMiscUser.C

```diff
@@ -2898,50 +2898,76 @@ uint32_t formatEcidString( ecmdDataBuffer & i_ecidBuffer, std::string  & o_wafer
 {
   uint32_t rc = ECMD_SUCCESS;
 
-  ecmdDataBuffer waferID(60);
-  char           waferIdText[11];
+  ecmdDataBuffer waferID;
+  std::string waferIdText;
+  uint32_t ecidPos = 0;
+
+  /* Need to query DLL product*/
+  ecmdDllInfo info;
+  rc = ecmdQueryDllInfo(info);
+  if (rc) {
+    ecmdOutputError("formatEcidString - Problems occurred trying to get Dll Info\n");
+    return rc;
+  }
+
+  if (info.dllProduct == ECMD_DLL_PRODUCT_P10)
```

Comment:
Kevin made comment today that the ECID change is driven by the Samsung wafers, so everything going forward would need to make use of the new format.

Should the format be passed into this function instead of basing it on the dllProduct?

Reply:
I guess this would drive new Hidden eCMD APIs for getEcid and getEcidVerbose.

Reply:
That would be my suggestion.. the format could come back with the data, and then these output functions can use that format field.